### DISCUSSION TITLE
Set Gray Level on ST75256

### DISCUSSION THREAD

**Discussion by SuperNinja73** [olikraus/u8g2#1742] [Ideas]
![image](https://user-images.githubusercontent.com/53425740/148290776-15cc5536-70fd-4de2-bdc0-5786531ed2bd.png)
Hi, 
page 70 of the ST75256 datasheet , it is capable of several shades of gray. I did not find the function in your excellent library. Do you plan to add it ?
I want to do something like this : ![modifier-parametres-rapides](https://user-images.githubusercontent.com/53425740/148290606-5f410ff0-dc66-4235-a25d-f2ee43293dfc.jpg)
Thanks

**Comment by olikraus** [comment]
U8g2 is a monochrome library. It does not support gray levels. In fact it had been an early design decision of this library not to support gray levels.

ST75256 based displays are used in monochome mode: 
https://github.com/olikraus/u8g2/blob/47e570c263e86562cca5956d89bdd6bf8745570a/csrc/u8x8_d_st75256.c#L320

**Reply by SuperNinja73** [reply]
``` 
 //U8X8_C( 0x031 ),				/* select 01 commands */
  U8X8_C( 0x020 ),				/* gray levels */
  U8X8_A( 0x01 ),
  U8X8_A( 0x03 ),
...
```
a little higher we see gray level controls. Does that mean that it can be unlocked or is it too complicated, and that we will have to wait for a U8g3 version? lol

**Comment by olikraus** [comment]
> Does that mean that it can be unlocked or is it too complicated

You can unlock gray level mode, but you still can't draw gray levels because there are no such functions ;-)

> U8g3 version?

It would be more like a complete new library with gray level support: All drawing functions must be extended to support shades of gray.

At least from my end, there is no plan to create such a library :-/

**Reply by SuperNinja73** [reply]
ok, i would settle for the enormous work you have already done. Thank you
